fiftyone/core/sample.py

```diff
@@ -464,6 +464,9 @@ def save(self):
                 "Cannot save a sample that has not been added to a dataset"
             )
 
+        if self._dataset.allow_sample_save_batching:
+            return self._dataset.batch_save_sample(self)
+
```

Comment:
`self.frames.save()` still needs to be called here, or else any edits to the frames of video samples won't be saved here!